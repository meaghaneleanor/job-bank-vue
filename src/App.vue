<template>
  <div id="App">
    <Header />
    <router-view class="grid-container" /> 
    <Footer />
  </div>
</template>

<script>
import Header from './components/common/Header';
import Footer from './components/common/Footer';

export default {
  name: 'App',
  components: {
    Header,
    Footer
  },
};
</script>
