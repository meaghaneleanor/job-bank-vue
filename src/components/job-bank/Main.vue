<template>
  <div>
    <h1>Ontario job bank</h1>
    <JobResult />
  </div>
</template>

<script>
import JobResult from './JobResult.vue';

export default {
  components: {
    JobResult
  }    
};
</script>

