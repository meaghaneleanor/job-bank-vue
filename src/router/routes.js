import routeTranslations from '../translations/route-translations.js';

import Main from '../components/job-bank/Main.vue';
import Page404 from '../components/common/Page404.vue';

/*
  Because the route paths are defined in English, the search
  for the route translations will always be English, to retrieve the
  alias path in french
*/
const translatedAlias = function(path) {
  return routeTranslations['en'][path];
};

export default [
  {
    name: 'job-bank-search',
    path: '/page/ontario-job-bank',
    alias: translatedAlias('/page/ontario-job-bank'),
    component: Main
  },
  {
    name: 'job-bank-search-results',
    path: '/page/ontario-job-bank/search-results',
    alias: translatedAlias('/page/ontario-job-bank/search-results'),
    component: Main,
  },
  {
    name: 'page-not-found',
    path: '/*',
    component: Page404
  }
];
