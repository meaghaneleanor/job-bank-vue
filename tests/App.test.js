import { mount, createLocalVue } from '@vue/test-utils';
import {$t} from './test.utils.js';
import VueRouter from 'vue-router';

import routes from '../src/router/routes';

import App from '../src/App.vue';
import Main from '../src/components/job-bank/Main.vue';

const localVue = createLocalVue();
const router = new VueRouter({
  mode: 'history',
  routes,
  options: { routes }
});
localVue.use(router);

const createComponent = mount(App, {
  mocks: {
    $t,
    translateRoute: jest.fn()
  },
  localVue,
  router,
  stubs: ['Header', 'Footer', 'router-view']
});


describe("App.vue", () => {
  describe('snapshot', () => {
    it('should display the expected html', () => {
      const wrapper = createComponent;
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
  
  describe.skip('router-view', () => {
    it("renders a child component via routing", () => {
      const wrapper = createComponent;
      wrapper.vm.$router.push("/page/ontario-job-bank");
      expect(wrapper.find(Main).exists()).toBe(true);
    });
  });
});
