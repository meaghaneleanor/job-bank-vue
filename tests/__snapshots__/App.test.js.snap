// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue snapshot should display the expected html 1`] = `"<div id=\\"App\\"><header-stub></header-stub> <router-view-stub class=\\"grid-container\\"></router-view-stub> <footer-stub></footer-stub></div>"`;
